### DISCUSSION TITLE
Error with Compiling

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#24902] [Q&A General]
Dear MOOSEers,

When I compile my application, the error always pop up, as shown in the screenshot, do u have any idea of what's the reason of it? Thanks. 

![Screenshot_20230705_135126](https://github.com/idaholab/moose/assets/79764031/52eff42c-3376-4e70-b94f-f95b652caab4)

**Comment by GiudGiud** [comment]
Can you please attach the output of running the diagnostics script in moose/scripts

**Reply by rh201** [reply]
do you mean this one? 
[libmesh_diagnostic.log](https://github.com/idaholab/moose/files/11958608/libmesh_diagnostic.log)

**Reply by milljm** [reply]
When I cat/grep through your diagnostic log, I am not finding any hits for "20.02.5". I am not sure how that path is being made relevant during `make` of `hound`.

However, I am finding the following in the log relating to slurm:

```pre
/usr/bin/gcc -I/shared/apps/.gcc/8.3/openmpi/4.0.5/include -fexceptions -pthread -L/shared/apps/.gcc/8.3/hwloc/2.4.1/lib -L/opt/slurm/current/lib -Wl,-rpath -Wl,/shared/apps/.gcc/8.3/hwloc/2.4.1/lib -Wl,-rpath -Wl,/opt/slurm/current/lib -Wl,-rpath -Wl,/shared/apps/.gcc/8.3/openmpi/4.0.5/lib -Wl,--enable-new-dtags -L/shared/apps/.gcc/8.3/openmpi/4.0.5/lib -lmpi
```

Specifically:
`-L/opt/slurm/current/lib`

I think perhaps some part of MOOSE was built while on a different machine, where `/opt/slurm/20.02.5` _was_ available, but no longer. Did you perchance build some of the MOOSE dependencies/MOOSE itself while on a headnode for your cluster?

**Reply by rh201** [reply]
Thanks for reply. I cannot fully understand, I do build MOOSE in  the home folder on cluster, in parallel with other apps.

**Reply by milljm** [reply]
Something 'changed' from when you built PETSc, to when you built libMesh, to when you built MOOSE, to when you built hound. During one of those steps, `/opt/slurm/20.02.5` was available... and then became unavailable.

**Reply by rh201** [reply]
Thanks. I cleaned the home folder and reinstalled MOOSE, but there was error after installed Petsc and do the check as below. what's the problem here
![屏幕截图 2023-07-06 154857](https://github.com/idaholab/moose/assets/79764031/bb784427-e607-470c-ac50-8c2c3e0128c2)

**Reply by GiudGiud** [reply]
Does not look like a big "diff" to me.
Can you paste text under triple quotes instead of screen shots please

Does the rest of the installation work?

**Reply by rh201** [reply]
OK, I'll do copy-paste.
So it puzzles me, as I followed the protocol on the MOOSE website, but it does not proceed correctly, even at petsc stage. The rest also does not work.
BTW, the installation protocol is changing now and then, what's the reason for that?

**Reply by GiudGiud** [reply]
We adapt the protocols as we find better ways to explain and more resilient installation methods, nothing wrong there
